src/components/common/header/FilterBar.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+
+import RainIcon from "../../../assets/svgs/climate/Rain.svg?react";
+import TemperatureIcon from "../../../assets/svgs/climate/Temperature.svg?react";
+import TornadoIcon from "../../../assets/svgs/climate/Tornado.svg?react";
+import DustIcon from "../../../assets/svgs/climate/Dust.svg?react";
+import DroughtIcon from "../../../assets/svgs/climate/Drought.svg?react";
+import SeaLevelIcon from "../../../assets/svgs/climate/SeaLevel.svg?react";
+
+import { rowFlex } from "../../../styles/flexStyles";
+import { useSearchParams } from "react-router-dom";
+
+const filters = [
+  { id: "rain", icon: RainIcon },
+  { id: "temperature", icon: TemperatureIcon },
+  { id: "tornado", icon: TornadoIcon },
+  { id: "dust", icon: DustIcon },
+  { id: "drought", icon: DroughtIcon },
+  { id: "seaLevel", icon: SeaLevelIcon },
+];
+
+const FilterBar = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const initialFilter = searchParams.get("filter");
+  const [activeFilter, setActiveFilter] = useState<string | null>(
+    initialFilter
+  );
+
+  useEffect(() => {
+    const newParams = new URLSearchParams(searchParams);
+    if (activeFilter) {
+      newParams.set("filter", activeFilter);
+    } else {
+      newParams.delete("filter");
+    }
+    setSearchParams(newParams);
+  }, [activeFilter]);
+
+  return (
+    <FilterConatiner>
+      {filters.map((filter) => {
+        const Icon = filter.icon;
+        return (
+          <IconWrapper
+            key={filter.id}
+            $active={activeFilter === filter.id}
+            onClick={() =>
+              setActiveFilter(activeFilter === filter.id ? null : filter.id)
+            }
+          >
+            <Icon />
+          </IconWrapper>
+        );
```

Comment:
```suggestion
{filters.map(({ id, icon: Icon }) => {
        return (
          <IconWrapper
            key={id}
            $active={activeFilter === id}
            onClick={() => setActiveFilter(activeFilter === id ? null : id)}
          >
            <Icon />
          </IconWrapper>
        );
```
이렇게 객체 구조 분해를 이용하면 좀 더 깔끔해질 것 같아요!